src/sn_utils.c

```diff
@@ -836,20 +882,27 @@ static int process_udp(n2n_sn_t * sss,
 
       memcpy(&(ack.cookie), &(reg.cookie), sizeof(n2n_cookie_t));
       memcpy(ack.edgeMac, reg.edgeMac, sizeof(n2n_mac_t));
+	    if ((reg.dev_addr.net_addr == 0) || (reg.dev_addr.net_addr == 0xFFFFFFFF) || (reg.dev_addr.net_bitlen == 0) ||
+	        ((reg.dev_addr.net_addr & 0xFFFF0000) == 0xA9FE0000 /* 169.254.0.0 */)) {
+		    assign_one_ip_addr(sss, comm, &ipaddr);
+		    ack.dev_addr.net_addr = ipaddr.net_addr;
+		    ack.dev_addr.net_bitlen = ipaddr.net_bitlen;
+	    }
       ack.lifetime = reg_lifetime(sss);
 
       ack.sock.family = AF_INET;
       ack.sock.port = ntohs(sender_sock->sin_port);
       memcpy(ack.sock.addr.v4, &(sender_sock->sin_addr.s_addr), IPV4_SIZE);
 
       ack.num_sn=0; /* No backup */
-      memset(&(ack.sn_bak), 0, sizeof(n2n_sock_t));
 
       traceEvent(TRACE_DEBUG, "Rx REGISTER_SUPER for %s [%s]",
 		 macaddr_str(mac_buf, reg.edgeMac),
 		 sock_to_cstr(sockbuf, &(ack.sock)));
 
-      update_edge(sss, reg.edgeMac, comm, &(ack.sock), now);
+      if(memcmp(reg.edgeMac, &null_mac, N2N_MAC_SIZE) != 0){
+	      update_edge(sss, &reg, comm, &(ack.sock), now);
+      }
```

Comment:
This is a change in logic. Does the null_mac is the indicator for IP address assignment?